portal/server/utils/csv-utils.js

```diff
@@ -236,7 +236,8 @@ const extractCsvData = async (file) => {
       // Read the .xlsx file using exceljs
       const workbook = new ExcelJS.Workbook();
       await workbook.xlsx.load(file.buffer, { sheetStubs: true }).then(async () => {
-        const worksheet = workbook.worksheets[0]; // Assume the utilisation report data is on the first sheet
+        // An Excel file must contain at least 1 visible worksheet at all times.
+        const worksheet = workbook.worksheets.find((sheet) => sheet.state === 'visible');
```

Comment:
Can/should we add a log and/or return an error of some kind if there is no visible worksheet?

Seems otherwise we'll be creating an XLSX and proceeding with the rest of the journey when no data is available/visible?

Reply:
Find will return a worksheet object or undefined - in theory we should never hit the undefined case as we can't have an xlsx file without at least 1 visible worksheet, if we did there would be an error in the function call below this which will be caught by the try-catch block in this function.

We could add a check after this line to throw a more specific error though if you think that may be beneficial, e.g.:
```
if (!worksheet) {
  throw new Error('No visible worksheet found in the Excel file');
}
```

Reply:
Sounds good yes please @sw-mattw just thinking for debugging purposes, if we could have a `console.error` at least - not sure off the top of my head if throwing an error would bubble up and be handled gracefully, rather than crashing anything(?)

Reply:
No worries, I've added the error code block from the above comment. This should be caught by the try-catch within this function which calls `console.error` itself with the error object and doesn't re-throw - so this should all be okay thanks

Reply:
Nice one thanks @sw-mattw